packages/node/src/channels.ts

```diff
@@ -0,0 +1,387 @@
+      id: appInstanceId,
+      ...appInstanceState
+    };
+    await this.addAppInstanceProposal(channel, appInstance);
+    this.appInstanceIdToMultisigAddress[appInstanceId] =
+      channel.multisigAddress;
+    return appInstanceId;
+  }
+
+  async install(params: Node.InstallParams): Promise<AppInstanceInfo> {
+    const channel = await this.getChannelFromAppInstanceId(
+      params.appInstanceId
+    );
+    const appInstance: AppInstanceInfo = channel.proposedAppInstances![
+      params.appInstanceId
+    ];
+    await this.installAppInstance(channel, appInstance);
+    return appInstance;
+  }
+
+  // Methods for interacting with the store persisting changes to a channel
+
+  // getters
+
+  /**
+   * Returns a JSON object with the keys being the multisig addresses and the
+   * values being objects reflecting the channel schema described above.
+   */
+  async getAllChannels(): Promise<object> {
+    const channels = await this.store.get(this.multisigKeyPrefix);
+    if (!channels) {
+      console.log("No channels exist yet");
+      return {};
+    }
+    return channels;
+  }
+
+  /**
+   * Returns a JSON object matching the channel schema.
+   * @param multisigAddress
+   */
+  async getChannelFromStore(multisigAddress: Address) {
+    return await this.store.get(`${this.multisigKeyPrefix}/${multisigAddress}`);
+  }
+
+  // setters
+
+  async save(channel: Channel) {
+    await this.store.set(
+      `${this.multisigKeyPrefix}/${channel.multisigAddress}`,
+      channel
+    );
+  }
+
+  /**
+   * The app's installation is confirmed iff the store write operation
+   * succeeds as the write operation's confirmation provides the desired
+   * atomicity of moving an app instance from pending to installed.
+   * @param channel
+   * @param appInstance
+   */
+  async installAppInstance(channel: Channel, appInstance: AppInstanceInfo) {
+    delete channel.proposedAppInstances[appInstance.id];
+    channel.appInstances[appInstance.id] = appInstance;
+    await this.store.set(
+      `${this.multisigKeyPrefix}/${channel.multisigAddress}`,
+      channel
+    );
+  }
+
+  /**
+   * Adds the given proposed appInstance to a channel's collection of proposed
+   * app instances.
+   * @param channel
+   * @param appInstance
+   */
+  async addAppInstanceProposal(channel: Channel, appInstance: AppInstanceInfo) {
+    channel.proposedAppInstances[appInstance.id] = appInstance;
+    await this.store.set(
+      `${this.multisigKeyPrefix}/${
+        channel.multisigAddress
+      }/proposedAppInstances`,
+      channel.proposedAppInstances
+    );
+  }
+
+  // Utility methods
+
+  static canonicalizeAddresses(addresses: Address[]): string {
+    addresses.sort((addrA: Address, addrB: Address) => {
+      return new ethers.utils.BigNumber(addrA).lt(addrB) ? -1 : 1;
+    });
+    return ethers.utils.hashMessage(addresses.join(""));
+  }
+
+  static getMultisigAddress(owners: Address[]): Address {
+    // TODO: implement this using CREATE2
```

Comment:
confused about the naming of this. as it stands this function should be called `generateNewMultisigAddress` or something since `get.*` functions should be idempotent.

Reply:
You are correct.